src/Stratis.Bitcoin.Features.Consensus/StakeValidator.cs

```diff
@@ -169,7 +169,12 @@ public Target GetNextTargetRequired(IStakeChain stakeChain, ChainedHeader chaine
             }
 
             // This is used in tests to allow quickly mining blocks.
-            if (consensus.PosNoRetargeting)
+            if (!proofOfStake && consensus.PowNoRetargeting) 
+            {
+                this.logger.LogTrace("(-)[NO_POW_RETARGET]:'{0}'", lastPowPosBlock.Header.Bits);
+                return lastPowPosBlock.Header.Bits;
+            }
+            else if (proofOfStake && consensus.PosNoRetargeting)
```

Comment:
Remove the `else` it has no additional value here

Reply:
It adds some performance gain. If the first if condition evaluates to true the second one will not be executed.